### PR TITLE
fix: Correctly handle passwordless in MFA audit events

### LINKED ISSUE #37826: Passwordless audit events display undefined user
image

If it's not possible to discern the user responsible for the audit event due to the nature of passwordless we shouldn't include undefined in the messaging. If it's possible to get the user information we should populate the event appropriately.

### PR DESCRIPTION
**Pull request by codingllama**
Do the following fixes:

* Correctly render "mfa_auth_challenge.create" events without an user
* Record the passwordless user in "mfa_auth_challenge.validate" events

Fixes #37826.

Changelog: Handle passwordless in MFA audit events

### CODE REVIEW

**Comment by codingllama** [comment]
I think we just missed the passwordless corner-cases on #36953. FYI @Joerger.

File: lib/auth/auth.go [hunk 1]
```diff
@@ -6299,7 +6299,66 @@ func (a *Server) validateMFAAuthResponseForRegister(ctx context.Context, resp *p
...
+	// Read ClusterName for audit.
+	var clusterName string
+	if cn, err := a.GetClusterName(); err != nil {
+		log.WithError(err).Warn("Failed to read cluster name")
+		// err swallowed on purpose.
+	} else {
+		clusterName = cn.GetClusterName()
+	}
+
+	// Take the user from the authData if the user param is empty.
```
Comment (codingllama) [inline lib/auth/auth.go:6321]: The code here is mainly a move from below, minus the `user == ""` logic you see here.

Comment (codingllama) [reply lib/auth/auth.go:6321]: I opted for an "internal" func so we avoid shenanigans with defers reading named return variables, which can be harder to maintain over time.

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick reviews!

**Pull request merged**